### PR TITLE
Add oci url to the list of acceptable urls

### PR DESCRIPTION
**Pull request by momesgin**
### Summary
contributes to #12204 

### Checklist
- [ ] The PR is linked to an issue and the linked issue has a Milestone, or no issue is needed
- [ ] The PR has a Milestone 
- [ ] The PR template has been filled out
- [ ] The PR has been self reviewed 
- [ ] The PR has a reviewer assigned
- [ ] The PR has automated tests or clear instructions for manual tests and the linked issue has appropriate QA labels, or tests are not needed
- [ ] The PR has reviewed with UX and tested in light and dark mode, or there are no UX changes

### CODE REVIEW

File: shell/utils/validators/formRules/__tests__/index.test.ts [hunk 1]
```diff
@@ -1117,6 +1117,7 @@ describe('formRules', () => {
     const testCases = [
       ['', undefined],
       ['https://github.com/rancher/fleet-examples.git', undefined],
+      ['oci://test.rancher.io/charts/mychart', undefined],
```
Comment (richard-cox) [inline shell/utils/validators/formRules/__tests__/index.test.ts:1120]: can this be expanded to cover all types defined in https://ranchermanager.docs.rancher.com/v2.9/how-to-guides/new-user-guides/helm-charts-in-rancher/oci-repositories#add-an-oci-based-helm-chart-repository

Comment (richard-cox) [inline shell/utils/validators/formRules/__tests__/index.test.ts:1120]: can this be expanded to cover more url types? the regex has me scared

**Review by richard-cox — CHANGES_REQUESTED** [review]

File: shell/utils/validators/formRules/index.ts [hunk 2]
```diff
@@ -458,7 +458,7 @@ export default function(t: Translation, { key = 'Value' }: ValidationOptions): {
     return runValidators(val, [startHyphen('label'), endHyphen('label'), startDot('label'), endDot('label'), required]);
   };
 
-  const repo: Validator = (val) => val && !/((git|ssh|http(s)?)|(git@[\w\.]+))(\:(\/\/)?)([\w\.@\:/\-~]+)(\.git)?(\/)?/.test(val) ? t('validation.setting.repo') : undefined;
+  const repo: Validator = (val) => val && !/((git|ssh|oci|http(s)?)|(git@[\w\.]+))(\:(\/\/)?)([\w\.@\:/\-~]+)(\.git)?(\/)?/.test(val) ? t('validation.setting.repo') : undefined;
```
Comment (richard-cox) [inline shell/utils/validators/formRules/index.ts:461]: Tan made a good point in slack, fleet doesn't support oci for their _git repo_'s (helm repos though do)

**Comment by momesgin** [comment]
@richard-cox based on the decision to remove the repo validation for now, I'll close this PR

**Pull request closed without merging**